lang/cs/Org.Apache.REEF.IO/PartitionedData/IPartition.cs

```diff
@@ -32,9 +34,9 @@ public interface IPartition
         string Id { get; }
 
         /// <summary>
-        /// Opens the stream to the underlying partition.
+        /// Gives a pointer to the underlying partition.
         /// </summary>
-        /// <returns>A read-only stream.</returns>
-        Stream Open();
+        /// <returns>The pointer to the underlying partition</returns>
+        T GetPartitionPointer();
```

Comment:
I'm concerned about the name here, as pointers do exist in C#. Yet, the return type here is not a pointer in that sense. "Reference" won't work for similar reasons. How about "Handle"?